methods/basic.go

```diff
@@ -26,10 +29,61 @@ func BasicRequestWithBody(c *cli.Context, method string) (string, error) {
 		return "", fmt.Errorf("error getting file info for stdin fd: %w", err)
 	}
 	if (stat.Mode() & os.ModeCharDevice) == 0 {
-		body, err = io.ReadAll(os.Stdin)
+		body, err = ioutil.ReadAll(os.Stdin)
 		if err != nil {
 			return "", fmt.Errorf("error reading from stdin: %w", err)
 		}
+	} else if c.Bool("editor") {
+		var path string
+		var editor = os.Getenv("EDITOR")
+
+		if editor == "" {
+			// check OS
+			currentOs := runtime.GOOS
+
+			// based on OS find assign default editor
+			if currentOs == "linux" || currentOs == "darwin" {
+				editor = "nano"
+			} else if currentOs == "windows" {
+				editor = "notepad"
```

Comment:
I don't have a windows machine to test this, and I haven't really used windows since 2006-2007 so I'm just guessing here: but will this still find notepad without the file extension, or should it be `notepad.exe`?

Reply:
@gbmor I have tested it on windows machine too.

Reply:
@devblin OK cool! Good to know. I'll remember that for when I have to target Windows in the future :+1: